src/pages/billing/Project.js

```diff
@@ -140,28 +207,42 @@ const ProjectDetail = ({ project, project: { projectName, creationStatus }, bill
   const [expandedWorkspaceName, setExpandedWorkspaceName] = useState()
   const [sort, setSort] = useState({ field: 'email', direction: 'asc' })
   const [workspaceSort, setWorkspaceSort] = useState({ field: 'name', direction: 'asc' })
+  const [billingProject, setBillingProject] = useState(project)
 
   const signal = Utils.useCancellation()
 
   const adminCanEdit = _.filter(({ roles }) => _.includes(billingRoles.owner, roles), projectUsers).length > 1
 
+  const workspacesInProject = useMemo(() => _.flow(
+    _.map('workspace'),
+    _.filter({ namespace: billingProject.projectName })
+  )(workspaces), [billingProject, workspaces])
+
+  const groups = groupByBillingAccountStatus(billingProject, workspacesInProject)
+  const billingAccountsOutOfDate = !_.every(_.isEmpty, [groups.error, groups.updating])
```

Comment:
Might this be easier to parse for people who aren't familiar with the code? It avoids iterating, since there are only two collections to check.
```suggestion
  const billingAccountsOutOfDate = !(_.isEmpty(groups.error) && _.isEmpty(groups.updating))
```